app/assets/src/components/Samples.jsx

```diff
@@ -1708,6 +1712,125 @@ function TableDownloadDropdown({ project_id, parent }) {
   );
 }
 
+class BackgroundModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modalOpen: false,
+      name: "",
+      description: "",
+      sample_names: []
+    };
+    this.handleOpen = this.handleOpen.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.renderTextField = this.renderTextField.bind(this);
+    this.sample_ids = props.parent.state.selectedSampleIds;
+  }
+  handleOpen() {
```

Comment:
Why do you need a new request to get sample names? 
You can just pass them as attributes to the component as you loaded them to the table, right?

Reply:
Hmm, I can't just pass the names from the sample table to the component for lookup, because whenever the user clicks on a project, the sample table from the previous project is lost -- but some samples from that project may still be selected and need to be added to the new background too. 

What I could do is add the sample name into the state whenever a sample is selected, just like we currently add the sample ID to keep track of it... but I feel this makes the organization of the code less clear. 

The sample selection checkbox feature is shared between multiple buttons ("Compare" / "Make background" / potentially others in the future), so it is currently based on sample_id only (rather than adding all the attributes needed by all of the CohortButtons into the state). The "Compare" (heatmap) action then retrieves all the information it needs (names, top taxa, scores, etc) once you click -- I think it is most consistent if the "Background" button functions similarly.

Reply:
"the sample table from the previous project is lost -- but some samples from that project may still be selected" - we need to discuss this, makes no sense from a UX perspective.
"compare" leads to a different page, "create background" creates a modal. It is a very different behavior.

Reply:
Sure, let's discuss. The reason is, people need to be able to select samples from multiple projects. So selection needs to stick even if user switches projects.

Reply:
agreed that a sample_id => name mapping will be ideal. It can just be stored in a class variable. it doesn't have to be in the state.